### PR TITLE
Add Neg gradient

### PR DESCRIPTION
**Pull request by vnghia**
@saxenasaurabh 

Part of #42668

### CODE REVIEW

File: tensorflow/c/experimental/gradients/math_grad.cc [hunk 1]
```diff
@@ -201,6 +202,30 @@ class MatMulGradientFunction : public GradientFunction {
...
+
+    grad_outputs->resize(1);
+
+    // Grad for X
+    std::vector<AbstractTensorHandle*> neg_outputs(1);
+    std::string name = "Neg_Grad";
+    TF_RETURN_IF_ERROR(ops::Neg(ctx->ctx, {grad_inputs[0]},
+                                absl::MakeSpan(neg_outputs), name.c_str()));
+
+    (*grad_outputs)[0] = neg_outputs[0];
```
Comment (vnvo2409) [inline tensorflow/c/experimental/gradients/math_grad.cc:223]: @saxenasaurabh 
I don't know if we should pass `grad_outputs` directly to `ops::Neg`.

Comment (saxenasaurabh) [reply tensorflow/c/experimental/gradients/math_grad.cc:223]: Yeah, you can just use `grad_outputs` here.

File: tensorflow/python/framework/experimental/math_ops.py [hunk 2]
```diff
@@ -30,3 +30,8 @@ def add(a, b, name=None):
 def mat_mul(a, b, name=None):
   ctx = context.get_default()
   return _math_ops.mat_mul(ctx, a, b, name)
+
+
+def neg(a, name=None):
```
Comment (saxenasaurabh) [inline tensorflow/python/framework/experimental/math_ops.py:35]: Could we add a test for this as well? You can look at tests in `unified_api_test.py`.

**Review by saxenasaurabh — COMMENTED** [review]
Thanks! This looks great. Few minor comments.

**Comment by vnvo2409** [comment]
@saxenasaurabh 
I have a question. In `tensorflow/tensorflow/c/experimental/ops/math_ops.cc`, we are doing something like
https://github.com/tensorflow/tensorflow/blob/2345c09855f73f0a56184a06a72261d3eb792c75/tensorflow/c/experimental/ops/math_ops.cc#L97-L106

We just pass the input and get the output. But I can not find out what happens when we call `Execute()`. Could you show me the list of ops and their implementation ( something like `Neg(input) { return -input;}` ) please ? It will come in handy when I need to port one of those from the core to `tensorflow/tensorflow/c/experimental/ops/math_ops.cc`. Thank you

**Comment by saxenasaurabh** [comment]
> @saxenasaurabh
> I have a question. In `tensorflow/tensorflow/c/experimental/ops/math_ops.cc`, we are doing something like
> https://github.com/tensorflow/tensorflow/blob/2345c09855f73f0a56184a06a72261d3eb792c75/tensorflow/c/experimental/ops/math_ops.cc#L97-L106
> 
> We just pass the input and get the output. But I can not find out what happens when we call `Execute()`. Could you show me the list of ops and their implementation ( something like `Neg(input) { return -input;}` ) please ? It will come in handy when I need to port one of those from the core to `tensorflow/tensorflow/c/experimental/ops/math_ops.cc`. Thank you

It may be useful to look at this [guide](https://www.tensorflow.org/guide/create_op) on adding a new op. For `Neg`, the kernel  is registered [here](https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/kernels/cwise_op_neg_1.cc) which calls into [Eigen](http://google3/third_party/tensorflow/core/kernels/cwise_ops.h?l=860&rcl=325179961).

**Review by saxenasaurabh — APPROVED** [review]

**Pull request merged**